cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -237,79 +241,56 @@ struct DispatchHistogram
     // Scales samples to evenly-spaced bins
     struct ScaleTransform
     {
+        using CommonT = typename cuda::std::common_type<LevelT, SampleT>::type;
+
+        static constexpr bool PRECOMPUTE_RECIPROCAL =
+          cuda::std::is_floating_point<CommonT>::value;
```

Comment:
It is not, and it should not be per @griwes:

> `is_floating_point` asks a very specific question, which is: is this a built-in floating point type. `cuda::std::is_floating_point` will never answer "yes" for `__half`, because `__half` does not behave like a built-in floating point type. [...] [Fixing] this would (1) go against what the standard defines it to mean, and (2) would have a potential of breaking code that assumes that `is_floating_point` and `is_class` are never true at the same time.

I've updated this to incorporate your recent changes for `__half`